x/farm/client/cli/tx.go

```diff
@@ -207,6 +209,90 @@ $ %s tx %s harvest stake --from mykey
 	return cmd
 }
 
+func NewCmdSubmitFarmingPlanProposal() *cobra.Command {
+	cmd := &cobra.Command{
+		Use:   "farming-plan [proposal-file]",
+		Args:  cobra.ExactArgs(1),
+		Short: "Submit a farming plan proposal",
+		Long: strings.TrimSpace(
+			fmt.Sprintf(`Submit a farming plan proposal along with an initial deposit.
+The proposal details must be supplied via a JSON file.
+
+Example:
+$ %s tx gov submit-proposal farming-plan <path/to/proposal.json> --from=<key_or_address> --deposit=<deposit_amount>
+
+Where proposal.json contains:
+
+{
+  "title": "Farming Plan Proposal",
+  "description": "Let's start farming",
+  "create_plan_requests": [
+    {
+      "description": "New Farming Plan",
+      "farming_pool_address": "cosmos1mzgucqnfr2l8cj5apvdpllhzt4zeuh2cshz5xu",
+      "termination_address": "cosmos1mzgucqnfr2l8cj5apvdpllhzt4zeuh2cshz5xu",
+      "reward_allocations": [
+        {
+          "pair_id": 1,
+          "rewards_per_day": "100000000stake"
+        },
+        {
+          "pair_id": 2,
+          "rewards_per_day": "50000000stake"
+        }
+      ],
+      "start_time": "2022-01-01T00:00:00Z",
+      "end_time": "2023-01-01T00:00:00Z"
+    }
+  ],
+  "terminate_plan_requests": [
+    {
+      "pair_id": 1
+    },
+    {
+      "pair_id": 2
+    }
+  ]
+}
```

Comment:
There is a couple of issues with this JSON format. 

1. This example returns an error stating that it cannot unmarshal string into Go value because `reward_allocation` is `sdk.Coins` type.
2. This example doesn't seem to be a practical example as it requests to create new plans and terminate them. 
3. TerminatePlanRequest has a field named `PoolId`, which should be `PairId`.

I suggest you to have the following JSON format and verify this with test code.
```json
{
  "title": "Farming Plan Proposal",
  "description": "Let's start farming",
  "create_plan_requests": [
    {
      "description": "New Farming Plan",
      "farming_pool_address": "cosmos1mzgucqnfr2l8cj5apvdpllhzt4zeuh2cshz5xu",
      "termination_address": "cosmos1mzgucqnfr2l8cj5apvdpllhzt4zeuh2cshz5xu",
      "reward_allocations": [
        {
          "pair_id": 1,
          "rewards_per_day": [
            {
              "amount": "100000000",
              "denom": "stake"
            }
          ]
        }
      ],
      "start_time": "2022-01-01T00:00:00Z",
      "end_time": "2023-01-01T00:00:00Z"
    }
  ],
  "terminate_plan_requests": [
    {
      "pair_id": 1
    },
    {
      "pair_id": 2
    }
  ]
}
```

Reply:
You can reference the following code snippet.

`util_test.go`
```go
package cli_test

import (
	"fmt"
	"testing"

	"github.com/cosmos/cosmos-sdk/testutil"

	"github.com/cosmosquad-labs/squad/v3/app/params"
	"github.com/cosmosquad-labs/squad/v3/x/farm/client/cli"
)

func TestParsePrivateFixedPlan(t *testing.T) {
	okJSON := testutil.WriteToNewTempFile(t, `
{
  "title": "Farming Plan Proposal",
  "description": "Let's start farming",
  "create_plan_requests": [
    {
      "description": "New Farming Plan",
      "farming_pool_address": "cosmos1mzgucqnfr2l8cj5apvdpllhzt4zeuh2cshz5xu",
      "termination_address": "cosmos1mzgucqnfr2l8cj5apvdpllhzt4zeuh2cshz5xu",
      "reward_allocations": [
        {
          "pair_id": 1,
          "rewards_per_day": [
            {
              "amount": "100000000",
              "denom": "stake"
            }
          ]
        }
      ],
      "start_time": "2022-01-01T00:00:00Z",
      "end_time": "2023-01-01T00:00:00Z"
    }
  ],
  "terminate_plan_requests": [
    {
      "pair_id": 1
    },
    {
      "pair_id": 2
    }
  ]
}
`)

	encodingConfig := params.MakeTestEncodingConfig()

	plan, err := cli.ParseFarmingPlanProposal(encodingConfig.Marshaler, okJSON.Name())
	fmt.Println("err: ", err)
	fmt.Println("plan: ", plan.String())
}
```